marklogic-client-api/src/main/java/com/marklogic/client/dataservices/impl/IOEndpointImpl.java

```diff
@@ -327,6 +344,25 @@ void assignEndpointState(InputStream[] output) {
             }
             return output;
         }
+
+        private void checkCallContextQueue() {
+            if(this.callContexts != null || !this.callContexts.isEmpty())
+                throw new InternalError("CallContext queue not empty");
```

Comment:
Because the error condition that we're preventing is a null pointer, would the check be better as a test that this.callContext != null?